### DISCUSSION TITLE
golang version of baidu apollo cyber, work in process, support shm transport

### DISCUSSION THREAD

**Discussion by haormj** [ApolloAuto/apollo#15484] [Ideas]
I recently implemented a Go version of Baidu Apollo Cyber, and it's still under development. I hope it will be helpful to you.

https://github.com/haormj/cyber

**Comment by daohu527** [comment]
Glad to hear this news. It may be used in other systems because it is a distributed architecture. We can discuss further work in the discussion.

**Reply by haormj** [reply]
this is also under development, not for production.

DONE
  - [x] support timer component
  - [x] support data driven component 0-4
  - [x] support run component(only write by go) by `cyber mainboard -d ./xxx.dag` by golang plugin
  - [x] support service by shm
  - [x] support communicate with c++ cyber by shm

TODO
  - [ ] support rtps, but https://github.com/eProsima/Fast-DDS/ not have golang version. maybe use cgo
  - [ ] support service discovery
  - [ ] ...
